containers/ecr-viewer/integration/listEcrDataService.test.tsx

```diff
@@ -0,0 +1,420 @@
+
+      const result = await listEcrData(
+        startIndex,
+        itemsPerPage,
+        columnName,
+        direction,
+        testDateRange,
+      );
+
+      expect(result).toBeEmpty();
+    });
+
+    it("should return data when found", async () => {
+      await (db as any)
+        .insertInto("ecr_viewer.ecr_data")
+        .values({
+          eicr_id: "12345",
+          set_id: "123",
+          data_source: "DB",
+          fhir_reference_link: "",
+          eicr_version_number: "1",
+          patient_name_first: "Billy",
+          patient_name_last: "Bob",
+          patient_birth_date: new Date("2024-12-01T12:00:00Z"),
+          date_created: new Date("2024-12-01T12:00:00Z"),
+          report_date: new Date("2024-12-01T12:00:00Z"),
+        })
+        .returningAll()
+        .executeTakeFirstOrThrow();
+      // @ts-ignore TS2364
+      const startIndex = 0;
+      const itemsPerPage = 25;
+      const columnName = "date_created";
+      const direction = "DESC";
+      const actual: EcrDisplay[] = await listEcrData(
+        startIndex,
+        itemsPerPage,
+        columnName,
+        direction,
+        testDateRange,
+      );
+      expect(actual).toEqual([
+        {
+          date_created: "12/01/2024 7:00\u00A0AM\u00A0EST",
+          ecrId: "12345",
+          patient_date_of_birth: "12/01/2024",
+          patient_first_name: "Billy",
+          patient_last_name: "Bob",
+          patient_report_date: "12/01/2024 12:00\u00A0AM\u00A0EST",
+          reportable_conditions: [null],
+          rule_summaries: [null],
+          eicr_set_id: "123",
+          eicr_version_number: "1",
+        },
+      ]);
+    });
+
+    it("should get data from the fhir_metadata table", async () => {
+      // @ts-ignore TS2364
+      const startIndex = 0;
+      const itemsPerPage = 25;
+      const columnName = "date_created";
+      const direction = "DESC";
+      const actual: EcrDisplay[] = await listEcrData(
+        startIndex,
+        itemsPerPage,
+        columnName,
+        direction,
+        testDateRange,
+      );
+      expect(actual).toEqual([
+        {
+          date_created: "12/01/2024 7:00\u00A0AM\u00A0EST",
+          ecrId: "12345",
+          patient_date_of_birth: "12/01/2024",
+          patient_first_name: "Billy",
+          patient_last_name: "Bob",
+          patient_report_date: "12/01/2024 12:00\u00A0AM\u00A0EST",
+          reportable_conditions: [null],
+          rule_summaries: [null],
+          eicr_set_id: "123",
+          eicr_version_number: "1",
+        },
+      ]);
+    });
+  });
+
+  describe("listEcrDataService with SQL Server", () => {
+    beforeAll(async () => {
+      process.env.METADATA_DATABASE_TYPE = "postgres";
+      process.env.METADATA_DATABASE_SCHEMA = "core";
+      await buildCoreAlias();
+    });
+
+    afterAll(async () => {
+      delete process.env.METADATA_DATABASE_TYPE;
+      await dropCoreAlias();
+    });
+    beforeEach(() => {
```

Comment:
this dance of setting metadata database types is hard to follow - do we really need all of these?